src/decode.rs

```diff
@@ -4964,8 +4960,10 @@ pub unsafe fn rav1d_submit_frame(c: &mut Rav1dContext) -> Rav1dResult {
                 first,
                 u32::MAX,
             );
-            if c.task_thread.cur != 0 && c.task_thread.cur < c.n_fc {
-                c.task_thread.cur -= 1;
+            if c.task_thread.cur.load(Ordering::Relaxed) != 0
+                && c.task_thread.cur.load(Ordering::Relaxed) < c.n_fc
+            {
+                c.task_thread.cur.fetch_sub(1, Ordering::Relaxed);
```

Comment:
Is the `Ordering::Relaxed` because it previously wasn't atomic and so shouldn't really need atomic synchronization, and `Ordering::Relaxed` is the least strict `Ordering`?  Previously (e.x. for `Abcd` in #662) I just used `Ordering::SeqCst` as the default for all atomics.  Should I do `Ordering::Relaxed` for these kinds on previously not atomic operations instead?

Reply:
Exactly. If the field isn't actually concurrently accessed it doesn't need any ordering and Relaxed should be the fastest lowering. We shouldn't be adding an ordering requirement where there was previously not one.

Reply:
I opened #678 to make `Abcd` do the same.